pkg/services/publicdashboards/api/middleware_test.go

```diff
@@ -1,105 +1,159 @@
 	"github.com/grafana/grafana/pkg/models"
-	"github.com/grafana/grafana/pkg/services/contexthandler/ctxkey"
-	fakeDatasources "github.com/grafana/grafana/pkg/services/datasources/fakes"
 	"github.com/grafana/grafana/pkg/services/publicdashboards"
-	publicdashboardsService "github.com/grafana/grafana/pkg/services/publicdashboards/service"
-	"github.com/grafana/grafana/pkg/services/query"
-	"github.com/grafana/grafana/pkg/setting"
+	. "github.com/grafana/grafana/pkg/services/publicdashboards/models"
+	"github.com/grafana/grafana/pkg/services/user"
 	"github.com/grafana/grafana/pkg/web"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 )
 
 func TestRequiresValidAccessToken(t *testing.T) {
-	t.Run("Returns 404 when access token is empty", func(t *testing.T) {
-		request, err := http.NewRequest("GET", "/api/public/ma/events/", nil)
-		require.NoError(t, err)
-
-		resp := runMiddleware(request, mockAccessTokenExistsResponse(false, nil))
-
-		require.Equal(t, http.StatusNotFound, resp.Code)
-	})
-
-	t.Run("Returns 200 when public dashboard with access token exists", func(t *testing.T) {
-		request, err := http.NewRequest("GET", "/api/public/ma/events/myAccessToken", nil)
-		require.NoError(t, err)
-
-		resp := runMiddleware(request, mockAccessTokenExistsResponse(true, nil))
-
-		require.Equal(t, http.StatusOK, resp.Code)
-	})
-
-	t.Run("Returns 400 when public dashboard with access token does not exist", func(t *testing.T) {
-		request, err := http.NewRequest("GET", "/api/public/ma/events/myAccessToken", nil)
-		require.NoError(t, err)
-
-		resp := runMiddleware(request, mockAccessTokenExistsResponse(false, nil))
-
-		require.Equal(t, http.StatusBadRequest, resp.Code)
-	})
-
-	t.Run("Returns 500 when public dashboard service gives an error", func(t *testing.T) {
-		request, err := http.NewRequest("GET", "/api/public/ma/events/myAccessToken", nil)
-		require.NoError(t, err)
-
-		resp := runMiddleware(request, mockAccessTokenExistsResponse(false, fmt.Errorf("error not found")))
+	tests := []struct {
+		Name                 string
+		Path                 string
+		AccessTokenExists    bool
+		AccessTokenExistsErr error
+		AccessToken          string
+		ExpectedResponseCode int
+	}{
+		{
+			Name:                 "Returns 404 when access token is empty",
+			Path:                 "/api/public/ma/events/",
+			AccessTokenExists:    false,
+			AccessTokenExistsErr: nil,
+			AccessToken:          "",
+			ExpectedResponseCode: http.StatusNotFound,
+		},
+		{
+			Name:                 "Returns 200 when public dashboard with access token exists",
+			Path:                 "/api/public/ma/events/myAccesstoken",
+			AccessTokenExists:    true,
+			AccessTokenExistsErr: nil,
+			AccessToken:          "myaccesstoken",
+			ExpectedResponseCode: http.StatusOK,
+		},
 
-		require.Equal(t, http.StatusInternalServerError, resp.Code)
-	})
+		// OWEN should this be a 404?
```

Comment:
@owensmallwood I think this should actually be a 404 instead of a 400 from a security perspective, however, I'm not entirely sure how this endpoint is used. Thoughts?

Reply:
I think either is fine. Both could be argued for. 

I didn't use a 404 bc I didn't think of it as a resource that couldn't be found. I thought of it more as a requirement for submitting a valid request to that endpoint. Doesn't matter to me which we use.

Reply:
I think this is a good practice. I'll look into token validation.